File: javascript/test/trajtrap.js
#!/usr/bin/env node
var liba = require("../liba")
function assert(expr, info) {
    if (!expr) {
        throw Error("assertion failed: " + info);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete()
    }
    if (liba.do_leak_check) {
        liba.do_leak_check()
    }
}
var ctx = new liba.trajtrap()
ctx.gen(2, 2, -2, 0, 4, 0, 0)
assert(ctx.t != undefined)
assert(ctx.p0 != undefined)
assert(ctx.p1 != undefined)
assert(ctx.v0 != undefined)
assert(ctx.v1 != undefined)
assert(ctx.vc != undefined)
assert(ctx.ta != undefined)
assert(ctx.td != undefined)
assert(ctx.pa != undefined)
assert(ctx.pd != undefined)
assert(ctx.ac != undefined)
assert(ctx.de != undefined)
console.log(ctx.pos(0.5), ctx.vel(0.5), ctx.acc(0.5))
do_leak_check(ctx)

File: java/test/_trajtrap.java
import liba.trajtrap;

class _trajtrap {
    public static void main(String[] args) {
        trajtrap obj = new trajtrap();
        obj.gen(2, 2, -2, 0, 4, 0, 0);
        System.out.println(obj.t());
        System.out.println(obj.p0());
        System.out.println(obj.p1());
        System.out.println(obj.v0());
        System.out.println(obj.v1());
        System.out.println(obj.vc());
        System.out.println(obj.ta());
        System.out.println(obj.td());
        System.out.println(obj.pa());
        System.out.println(obj.pd());
        System.out.println(obj.ac());
        System.out.println(obj.de());
        System.out.println(obj.pos(0.5) + "," + obj.vel(0.5) + "," + obj.acc(0.5));
    }
}
